L97-python-setuptools-plugin.md

```diff
@@ -0,0 +1,170 @@
+L97 - Python: Setuptools entrypoint for generating file
+----
+* Author(s): Patrick Lannigan
+* Approver: lidizheng and gnossen
+* Status: Draft
+* Implemented in: Python
+* Last updated: 2022-02-18
+* Discussion at: https://groups.google.com/g/grpc-io/c/H617sSzoV-8
+
+## Abstract
+
+Implement a `setuptools` entrypoint for generating files when a Python distribution is being built.
+This type of integration would be called automatically by `setuptools` and would not require and
+explicit call by the project to generate the files. The entrypoint would be configured using an
+established python project configuration file and would only require the `grpcio-tools` package to be
+listed as a build time dependency.
+
+## Background
+
+Currently `grpc_tools` [provides a command][grpc_command] for integrating with `setuptools`. However,
+this requires custom integration on the part of the package (example:
+[grpc_health_checking][grpc_command_example_integration] and
+[explicitly calling the command][grpc_command_example_call] when building the wheel).
+
+With the introduction of [PEP 517][pep_517] and [PEP 518][pep_518], directly calling `setup.py` will
+start becoming less common. Additionally, the use of custom commands will not work with projects that
+use this mechanism and solely rely on the [declarative config in `setup.cfg`][declarative_config],
+which was added to `setuptools` in [v30.3.0][release_declarative_config] (Dec. 2016). Though, use of
+the functionality defined in this proposal does not require a project to use the declarative config.
+
+Custom commands are not the only integration point that is available. `setuptools` also supports
+entrypoints that allows for packages to act as a plugin and provide custom behavior. This is
+[documented with examples][setuptools_entrypoint] of generating files based on revision control
+information. This could be used to generate source files based on `.proto` files.
+
+[Original Feature Request][original_feature_request]
+
+## Proposal
+
+`grpcio-tools` will register a new entrypoint function to respond to the `setuptools.file_finders`
+hook. This hook will generate Python files based on the `.proto` files and tell `setuptools where
+they were generated.
```

Comment:
Can we update the example to show how the proto to Python generation should work?